src/MongoDB.Driver/MongoClient.cs

```diff
@@ -417,6 +461,13 @@ private IReadWriteBindingHandle CreateReadWriteBinding(IClientSessionHandle sess
             return new ReadWriteBindingHandle(binding);
         }
 
+        private ChangeStreamOperation<TResult> CreateChangeStreamOperation<TResult>(
+            PipelineDefinition<ChangeStreamDocument<BsonDocument>, TResult> pipeline,
+            ChangeStreamOptions options)
+        {
+            return ChangeStreamHelper.CreateChangeStreamOperation(pipeline, options, _settings.ReadConcern, GetMessageEncoderSettings());
```

Comment:
The hard work of creating ChangeStreamOperations is now in a helper class where the three overloads of CreateChangeStreamOperation (for client, database and collection) can share some code.